tests/transport/test_grpc.py

```diff
@@ -58,6 +59,8 @@ def test_call_no_refresh(self):
 
         plugin(context, callback)
 
+        time.sleep(2)
```

Comment:
Are these timeouts required for the tests to run?

Reply:
No, they are not required; I added them so that the thread pool has some (more) time to finish the work in the rare case that the main thread performs the check before the callback is actually called.